easybuild/scripts/bootstrap_eb.py

```diff
@@ -221,37 +226,71 @@ def stage0(tmpdir):
     else:
         debug("Found setuptools in expected path, good!")
 
-def stage1(tmpdir):
+    return distribute_egg_dir
+
+
+def stage1(tmpdir, sources_path):
     """STAGE 1: temporary install EasyBuild using distribute's easy_install."""
 
     info("\n\n+++ STAGE 1: installing EasyBuild in temporary dir with easy_install...\n\n")
 
+    # determine locations of source tarballs, if sources path is specified
+    source_tarballs = []
+    if sources_path is not None:
+        info("Fetching sources from %s..." % sources_path)
+        for pkg in ['vsc-base', 'easybuild-framework', 'easybuild-easyblocks', 'easybuild-easyconfigs']:
+            pkg_tarball_glob = os.path.join(sources_path, '%s*.tar.gz' % pkg)
+            pkg_tarball_paths = glob.glob(pkg_tarball_glob)
+            if len(pkg_tarball_paths) > 1:
+                error("Multiple tarballs found for %s: %s" % (pkg, pkg_tarball_paths))
+            elif len(pkg_tarball_paths) == 0:
+                if pkg != 'vsc-base':
+                    error("Missing source tarball: %s" % pkg_tarball_glob)
+            else:
+                info("Found %s" % pkg_tarball_paths[0])
+                source_tarballs.append(pkg_tarball_paths[0])
+
     from setuptools.command import easy_install
 
     # prepare install dir
     targetdir_stage1 = os.path.join(tmpdir, 'eb_stage1')
     prep(targetdir_stage1)  # set PATH, Python search path
 
-
     # install latest EasyBuild with easy_install from PyPi
     cmd = []
     cmd.append('--upgrade')  # make sure the latest version is pulled from PyPi
     cmd.append('--prefix=%s' % targetdir_stage1)
-    cmd.append('easybuild')
+
+    if sources_path:
+        # install provided source tarballs
+        cmd.extend(source_tarballs)
+    else:
+        # install meta-package easybuild from PyPI
+        cmd.append('easybuild')
+
     if not print_debug:
         cmd.insert(0, '--quiet')
-    debug("installing EasyBuild with 'easy_install %s'" % (" ".join(cmd)))
+    info("installing EasyBuild with 'easy_install %s'" % (' '.join(cmd)))
     easy_install.main(cmd)
 
     # clear the Python search path, we only want the individual eggs dirs to be in the PYTHONPATH (see below)
     # this is needed to avoid easy-install.pth controlling what Python packages are actually used
     os.environ['PYTHONPATH'] = ''
 
-    versions = {}
+    # template string to inject in template easyconfig
+    templates = {}
 
     pkg_egg_dir_framework = None
-    for pkg in ['easyconfigs', 'easyblocks', 'framework']:
-        pkg_egg_dir = find_egg_dir_for(targetdir_stage1, 'easybuild-%s' % pkg)
+    for pkg in ['vsc-base', 'easybuild-framework', 'easybuild-easyblocks', 'easybuild-easyconfigs']:
+        templates.update({pkg: ''})
+
+        pkg_egg_dir = find_egg_dir_for(targetdir_stage1, pkg)
+        if pkg_egg_dir is None:
+            if pkg == 'vsc-base':
+                # vsc-base is optional in older EasyBuild versions
```

Comment:
no, we need a single bootstrap for both versions (and at this point, we don't know the version being installed yet)